src/warp.jl

```diff
@@ -0,0 +1,38 @@
+# None of these types are provided by this package, so the following line seems unclean
+#     @inline Base.getindex(A::AbstractExtrapolation, v::StaticVector) = A[convert(Tuple, v)...]
+# furthermore it would be ambiguous with getindex(::Extrapolation, xs...) after https://github.com/tlycken/Interpolations.jl/pull/141
+@inline _getindex(A, v::StaticVector) = A[convert(Tuple, v)...]
+
+warp(img::AbstractArray, args...) = warp(interpolate(img, BSpline(Linear()), OnGrid()), args...)
+
+function warp{T<:Colorant,S}(::Type{T}, img::AbstractArray{S}, args...)
+    TCol = typeof(convert(T,one(S)))
```

Comment:
Not quite. I do this only for user convenience, so they don't need to specify fully qualified Colorants

```julia
julia> T = Gray
ColorTypes.Gray{T<:Union{AbstractFloat,Bool,FixedPointNumbers.FixedPoint}}

julia> S = RGB{N0f8}
ColorTypes.RGB{FixedPointNumbers.Normed{UInt8,8}}

julia> TCol = typeof(convert(T,one(S)))
ColorTypes.Gray{FixedPointNumbers.Normed{UInt8,8}}
```